File: languages/curso-java/programacion-estructurada/estructuras-control/bucles/BreakAndContinue/BreakAndContinue.java
public class BreakAndContinue {
    public static void main(String[] args) {
        // Break - interrumpe el bucle
        System.out.println("--- Break ---");
        for (int i = 1; i <= 10; i++) {
            if (i == 6) {
                System.out.println("Break en i = " + i);
                break;
            }
            System.out.println("i = " + i);
        }

        // Continue - salta a la siguiente iteracion
        System.out.println("\n--- Continue (pares) ---");
        for (int i = 1; i <= 10; i++) {
            if (i % 2 != 0) continue;  // Salta impares
            System.out.print(i + " ");
        }
        System.out.println();

        // Break con etiqueta (bucles anidados)
        System.out.println("\n--- Break con etiqueta ---");
        externo:
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                if (i == 1 && j == 1) break externo;
                System.out.println("i=" + i + ", j=" + j);
            }
        }
    }
}

File: languages/curso-js/programacion-estructurada/breakAndContinue.js
/*
ARCHIVO: breakAndContinue.js
TEMA: Control de saltos en bucles (Break y Continue)
En este ejemplo se utilizan:
- Bucle 'for...in' para recorrer índices de un string
- Sentencia 'break': Detiene el bucle por completo
- Sentencia 'continue': Salta la iteración actual y sigue con la siguiente
*/

let palabra = "JavaScript";

// Ejemplo de BREAK: Se detiene al llegar a la posición 4
for (let f in palabra) {
    document.write(f);
    if (f == 4) {
        document.write("<br>Estamos en la posicion 4 <br>");
        break; // Sale del bucle inmediatamente
    }
}

// Ejemplo de CONTINUE: Filtra las vocales
for (let f in palabra) {
    // Si la letra es una vocal, se salta el document.write
    if(palabra[f] == "a" || palabra[f] == "e" || palabra[f] == "i" || palabra[f] == "o" || palabra[f] == "u") {
        continue; 
    }
    document.write(palabra[f]); // Solo escribirá las consonantes
}